File: code/cpp/ch19/practice/practice_01_shortest_path.cpp
/*
 * Practice 1: Shortest Path (Unweighted)
 * Chapter 19: Graphs I — Exploring Networks
 *
 * PROBLEM: Find shortest distances from source to all nodes. -1 if unreachable.
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
#include <iostream>
#include <queue>
#include <vector>
using namespace std;

vector<int> solve(int n, vector<vector<int>>& edges, int source) {
    // TODO: Replace this with your solution
    return vector<int>(n, -1);
}

int main() {
    int n, m, source; cin >> n >> m >> source;
    vector<vector<int>> edges(m, vector<int>(2));
    for (int i = 0; i < m; i++) cin >> edges[i][0] >> edges[i][1];
    auto dist = solve(n, edges, source);
    for (int i = 0; i < n; i++) cout << dist[i] << " ";
    cout << endl;
    return 0;
}

File: code/java/ch19/practice/Practice01ShortestPath.java
package ch19.practice;

import java.util.*;

/**
 * Practice 1: Shortest Path (Unweighted)
 * Chapter 19: Graphs I — Exploring Networks
 *
 * PROBLEM: Find shortest distances from source to all nodes. -1 if unreachable.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Practice01ShortestPath {
    public static int[] solve(int n, int[][] edges, int source) {
        // TODO: Replace this with your solution
        int[] dist = new int[n];
        Arrays.fill(dist, -1);
        return dist;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt(), m = sc.nextInt(), source = sc.nextInt();
        int[][] edges = new int[m][2];
        for (int i = 0; i < m; i++) { edges[i][0] = sc.nextInt(); edges[i][1] = sc.nextInt(); }
        System.out.println(Arrays.toString(solve(n, edges, source)));
        sc.close();
    }
}
